internal/controller/databaseclusterrestore_controller.go

```diff
@@ -668,7 +668,18 @@ func (r *DatabaseClusterRestoreReconciler) ReconcileWatchers(ctx context.Context
 
 	log := log.FromContext(ctx)
 	addWatcher := func(dbEngineType everestv1alpha1.EngineType, obj client.Object) error {
-		src := source.TypedKind(r.Cache, obj, handler.EnqueueRequestForOwner(r.Scheme, r.RESTMapper(), &everestv1alpha1.DatabaseCluster{}))
```

Comment:
This line here is the actual issue - it should have been `&everestv1alpha1.DatabaseClusterRestore{}`